hnn_core/network.py

```diff
@@ -35,27 +37,51 @@ def read_spikes(fname, gid_dict=None):
     spike_times = []
     spike_gids = []
     spike_types = []
+    spike_tstart = []
+    spike_tstop = []
     for file in sorted(glob(fname)):
         spike_trial = np.loadtxt(file, dtype=str)
         spike_times += [list(spike_trial[:, 0].astype(float))]
         spike_gids += [list(spike_trial[:, 1].astype(int))]
 
         # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
-        # spike type. If reading a legacy version, validate that a gid_dict is
-        # provided.
-        if spike_trial.shape[1] == 3:
+        # spike type, or a 4th and 5th column for tstart and tstop. If reading
+        # a legacy version, validate that gid_dict, tstart, and tstop provided.
+        if spike_trial.shape[1] >= 3:
             spike_types += [list(spike_trial[:, 2].astype(str))]
-        else:
+        elif spike_trial.shape[1] == 2:
             if gid_dict is None:
                 raise ValueError("gid_dict must be provided if spike types "
                                  "are unspecified in the file %s" % (file,))
             spike_types += [[]]
 
-    spikes = Spikes(times=spike_times, gids=spike_gids, types=spike_types)
+        if spike_trial.shape[1] == 5:
+            spike_tstart += np.unique(spike_trial[:, 3].astype(float)).tolist()
+            spike_tstop += np.unique(spike_trial[:, 4].astype(float)).tolist()
```

Comment:
if the user provides a `tstart` / `tstop` here it will be silently ignored which could be surprising. What's the desired behavior? Should you raise an error? Or always treat the user provided one with higher priority?

Reply:
I agree, it's not clear to me if overriding information from the file is a realistic use case. If the user really cares they can edit the class post-hoc. I've added a test to raise an error for now.

Reply:
Maybe the best way to handle this is to only raise an error if 1) tstart and tstop are provided in both the file and function call and 2) they are inconsistent.